learning_assistant/management/commands/set_course_prompts.py

```diff
@@ -0,0 +1,121 @@
+"""
+Django management command to generate course prompts.
+"""
+import json
+import logging
+from posixpath import join as urljoin
+
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from edx_rest_api_client.auth import SuppliedJwtAuth
+from edx_rest_api_client.client import USER_AGENT
+from opaque_keys.edx.keys import CourseKey
+
+try:
+    from openedx.core.djangoapps.oauth_dispatch.jwt import create_jwt_for_user
+except ImportError:
+    create_jwt_for_user = None
+
+from learning_assistant.models import CoursePrompt
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Django Management command to create a set of course prompts
+    """
+
+    def add_arguments(self, parser):
+
+        # list of course ids
+        parser.add_argument(
+            '--course_ids',
+            dest='course_ids',
+            help='comma separated list of course_ids to generate',
+        )
+
+        # pre-message
+        parser.add_argument(
+            '--pre_message',
+            dest='pre_message',
+            help='message to prepend to course topics',
+        )
+
+        parser.add_argument(
+            '--skills_descriptor',
+            dest='skills_descriptor',
+            help='message that describes skill structure'
+        )
+
+        # post-message
+        parser.add_argument(
+            '--post_message',
+            dest='post_message',
+            help='message to append to course topics',
+        )
+
+        # service username
+        parser.add_argument(
+            '-u',
+            '--username',
+            dest='username',
+            help='username for authenticating calls to discovery API'
+        )
+
+    @staticmethod
+    def _get_discovery_api_client(username):
+        """
+        Returns an API client which can be used to make Catalog API requests.
+        """
+        User = get_user_model()
```

Comment:
Interesting. Those changes eventually lead back to https://github.com/openedx/public-engineering/issues/37, which says, "We will be switching usages of the old client to OAuthAPIClient as part of this removal."

Ed Zeracor left a comment, "In some cases the request is on the behalf of the user and we are passing JOTs around. We will still need to do this, but the current client doesn’t have an interface for supporting JOTs", which I think was the case for the catalog service. I imagine that's why it was implemented this way.

I'd lean toward using the `OAuthAPIClient`, since it seems like the intended direction. It sounds like to me that the catalog service was implemented this way because the existing code wasn't well supported by `OAuthAPIClient`, but this is new code. Registrar has an example of calling discovery with `OAuthAPIClient` - it doesn't look any more complex.

What do you think?